fireplace/cards/utils.py

```diff
@@ -38,6 +40,43 @@
 AT_MAX_MANA = lambda s: MANA(s) == 10
 
 
+# take a list of weights and a list of card pools and produce a random weighted sample without replacement
+# len(weights) == len(card_sets) (one weight per card set)
+def weighted_card_choice(source, weights: List[int], card_sets: List[str], count: int):
+
+	# special case for single weight (generic random pick)
+	if len(card_sets)==1:
+		return [source.controller.card(card, source=source) for card in random.sample(card_sets[0], count)]
```

Comment:
I've changed the weight line to use set (len(set(weights)) == 1) but the two return statements aren't quite the same, AFAIK any other way of writing it will be more complicated or involve additional assignments or if statements. Please show me a reformulation if there is a Python way I don't know of :)